src/monster.cpp

```diff
@@ -2644,87 +2644,29 @@ void monster::process_effects()
 
 bool monster::make_fungus()
 {
-    if( is_hallucination() ) {
-        return true;
-    }
-    char polypick = 0;
-    const mtype_id &tid = type->id;
-    if( type->in_species( species_FUNGUS ) ) { // No friendly-fungalizing ;-)
+    if( is_hallucination() || type->in_species( species_FUNGUS ) ) {
         return true;
     }
-    if( !made_of( material_id( "flesh" ) ) && !made_of( material_id( "hflesh" ) ) &&
-        !made_of( material_id( "veggy" ) ) && !made_of( material_id( "iflesh" ) ) &&
-        !made_of( material_id( "bone" ) ) ) {
-        // No fungalizing robots or weird stuff (mi-gos are technically fungi, blobs are goo)
-        return true;
+
+    // if one of its materials is immune
+    // monster is immune
```

Comment:
Should having any immune material make a monster immune to fungus damage, even if it also has vulnerable materials?  It looks like the old code was checking the inverse -- only if a monster had no vulnerable materials would it bail out early.

Reply:
I added it this way because I think it will allow more flexible decisions for modders. In game you won't really note a difference; for example, since the prototype robot only material is steel it won't take damage or be stunned in both versions.

Reply:
Fair enough :+1: 
Wasn't sure how often multi-material monsters existed in vanilla, so the whole point may be moot